cpp/include/cugraph/algorithms.hpp

```diff
@@ -1606,6 +1606,31 @@ void core_number(raft::handle_t const& handle,
                  size_t k_last           = std::numeric_limits<size_t>::max(),
                  bool do_expensive_check = false);
 
+/**
+ * @brief   Extract K Core of a graph
+ *
+ * @throws     cugraph::logic_error when an error occurs.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
+ * @tparam weight_t Type of edge weights. Needs to be a floating point type.
+ * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
+ * @param  graph           cuGraph graph in coordinate format
+ * @param  k               Order of the core. This value must not be negative.
+ * @param  core_numbers    Output from core_number algorithm
+ *
+ * @return edge list for the graph
+ */
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
```

Comment:
Would it be a better idea to pass `bool transpose=false` as template parameter and then use transpose as a named parameter to the function instead of implicit false?

Reply:
We try to avoid excess compilation of different templated functions, as this increased compile time.

*Most* of our functions are implemented based on an assumption about whether the graph is stored transposed or not.  For example, `pagerank` assumes that `store_transposed=true`.  Implementing `pagerank` if `store_transposed=false` would result in a great deal more synchronization (and communication in a multi-gpu environment).  There are a few examples that will work on a graph in either orientation.

Supporting this would require providing an implementation that would work with either orientation of the graph.  It seems like as long as this matches the requirement for `core_number` (which has to be called first) then just one orientation is sufficient.